manager.go

```diff
@@ -376,12 +409,17 @@ func (m *Manager) dial(ctx context.Context) error {
 		dialOpts = append(dialOpts, rpc.WithInsecure())
 	}
 
-	conn, err := rpc.DialDirectGRPC(ctx, u.Host, m.logger, dialOpts...)
+	conn, err := rpc.DialDirectGRPC(ctx, u.Host, m.logger.AsZap(), dialOpts...)
 	if err != nil {
 		return err
 	}
 	m.conn = conn
 	m.client = pb.NewAgentDeviceServiceClient(m.conn)
+
+	// todo: ideally we would create the NetAppender ASAP (so logs are captured) and only *connect* it here.
+	if err := m.attachNetAppender(); err != nil {
+		m.logger.Errorw("error attaching NetAppender", "err", err)
```

Comment:
It feels like this should be returning the error here, not just logging it. What's the scenario you're expecting where this is non-fatal? From looking at the code, I think the ONLY possible error this could ever log would be if if it failed to get the hostname (from the os.Hostname() call inside NewNetAppender()) which I'd consider a pretty critical failure. All the others are impossible to hit per my above comment.

Reply:
this is an unattended device -- if it can possibly start up, it should

os.hostname could fail if procfs is unreadable https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/os/sys_linux.go;l=40

(or the implementation of netappender could change such that it returns errors in more cases)

Reply:
if we get to this part of the function, `m.client` is non-nil and a reattempt would short-circuit anyway (in the `m.client != nil` guard above)

I think this gets easier once we do NetAppender delayed start in https://viam.atlassian.net/browse/APP-5330 -- okay to keep as-is for now, then improve when the NetAppender change is done?

Reply:
Yeah, again, as long as it's done before we do OTHER new projects in agent (though no rush until then), I'm fine with that (and agree it'll be easier/cleaner.) I just don't want to let tech debt pile up.

Reply:
yup agree, bumped 5330 to high and will submit a netappender PR sometime this week